### DISCUSSION TITLE
CLI interface to export logbooks

### DISCUSSION THREAD

**Discussion by muihlinn** [foldynl/QLog#421] [Ideas]
It'll be nice to be able to use Qlog to export the logbook to ADIF (and possibly other formats) via CLI, making it possible to automate unattended backups and uploads to several online logbooks via cron without relying into "hacking" trough sqlite interaction.

**Comment by foldynl** [comment]
no need to hack anything. [Automatic backup](https://github.com/foldynl/QLog/wiki/Autobackup) performs the ADX dump at every start. Is this not enough to use?

**Reply by muihlinn** [reply]
> no need to hack anything. [Automatic backup](https://github.com/foldynl/QLog/wiki/Autobackup) performs the ADX dump at every start. Is this not enough to use?

that's perfectly fine for backups.  I'm not worried about backing up data, it's good enough as it is and I happily use it.

I'm looking for a canonical procedure to extract logbooks, or slicing the logbook from the database for further manipulation [for any purpose] without having to start Qlog to export a set manually. Often I do this remotely,  whithout the chance of x-forwarding or similar, if Qlog has't started after the last shack session in that machine, AFAIK part of the logbook still needs to be exported to those dumps. 

It doesn't have to be necessarily integrated in QLog (although desirable) any small utility will do it, and I even can lend a hand to write it. Will take me a time because it's have been ages since last time I digged into sqlite or cross platform stuff. 

Just wanted to know if something like this is in the roadmap, considered, other people points of view, etc.

**Reply by foldynl** [reply]
>>> I'm looking for a canonical procedure to extract logbooks, or slicing the logbook from the database for further manipulation [for any purpose] without having to start Qlog to export a set manually. Often I do this remotely, whithout the chance of x-forwarding or similar, if Qlog hasn't started after the last shack session in that machine, AFAIK part of the logbook still needs to be exported to those dumps.

these are exactly the reasons that lead to the fact that it should be written as an external utility. Qlog is a GUI Multiplatform application and must be expanded accordingly. If it will be an external application, then you can start designing from the beginning. I'm saying, you need to think about the fact that there are many forms of QLog (.deb package, Windows package, discontinued MacOS DMG, Flatpak, there is request to prepare SNAP and other installation packages). Each of these packages has its own specifications that must be taken into account when designing a new feature. If you need to somehow "customize" the code, then QLog offers you that it is open and you know exactly what is defined and how it works. That's the essence of open-source. However, I would not like to turn QLog into an application that contains "everything"  without making sense.  
I can imagine that it doesn't end with export, because it is necessary to define the profile for the exporter, the format for export, maybe even the columns for export... And we will have a very complex CLI interface here, which will be used by advanced users only.

>>> Just wanted to know if something like this is in the roadmap, considered, other people points of view, etc.

However, it is good to open this discussion. I could be wrong and most users want this feature.

It's just really important to think more about what platforms QLog is supported on and if the required feature will be widely used.  That's how I, for example, already denied the on-screen notifications that I would like to see in QLog. Unfortunately, these notifications are not very 'nice' on the Windows platform, so I had to leave this idea.

**Reply by muihlinn** [reply]
I'll take a look into it as soon as I have some spare time, at least as proof of concept.

**Comment by oh2gba** [comment]
And then use Déjà Dup Backups (or similar) to backup the ADX dumps. 
Pretty reliable I'd say. 

Files are stored: ~/.local/share/hamradio/QLog

Maybe some more control over where and what is stored? after using QLog for a little while, the directory has grown to:
du -sh QLog/
531M	QLog/

**Comment by foldynl** [comment]
Déjà Dup or other backup solution should be used regularly and for the entire home directory, in which Qlog data is also stored. If the user makes exceptions, it is his/her choice. Qlog has a clearly described backup policy and a directory where these backups are made. If this directory is not suitable, it is possible to use a symbolic link.

Gentlemen, it is necessary to keep in mind that there is Flatpak, Windows, which have own specific requirements for filesystem.
